extensions/amp-brightcove/1.0/component.js

```diff
@@ -0,0 +1,225 @@
+      const eventType = data?.event;
+      switch (eventType) {
+        case 'ready':
+          onReady?.(data, player);
+          break;
+        case 'playing':
+          onPlayingState?.(true);
+          break;
+        case 'pause':
+        case 'ended':
+          onPlayingState?.(false);
+          break;
+      }
+
+      if (data?.['ct']) {
+        playerStateRef.current['currentTime'] = data['ct'];
+      }
+      if (data?.['pr']) {
+        playerStateRef.current['playedRanges'] = data['pr'];
+      }
+      if (data?.['dur']) {
+        playerStateRef.current['duration'] = data['dur'];
+      }
+
+      if (events[eventType]) {
+        dispatchCustomEvent(currentTarget, events[eventType]);
+        return;
+      }
+
+      if (eventType === 'volumechange') {
+        const isMuted = data['muted'];
+        if (isMuted == null || isMuted == muted) {
+          return;
+        }
+        setMuted(isMuted);
+        dispatchCustomEvent(currentTarget, mutedOrUnmutedEvent(isMuted));
+      }
+    },
+    [muted, onReady, player, onPlayingState]
+  );
+
+  // Check for valid props
+  if (!checkProps(account)) {
+    return null;
+  }
+
+  return (
+    <VideoIframe
+      ref={ref}
+      {...rest}
+      autoplay={autoplay}
+      makeMethodMessage={makeMethodMessage}
+      muted={muted}
+      onIframeLoad={(event) => {
+        const {currentTarget} = event;
+        dispatchCustomEvent(currentTarget, 'canplay');
+      }}
+      onMessage={onMessage}
+      playerStateRef={playerStateRef}
+      src={src}
+    />
+  );
+}
+const Brightcove = forwardRef(BrightcoveWithRef);
+Brightcove.displayName = 'Brightcove'; // Make findable for tests.
+export {Brightcove};
+
+/**
+ * id is either a Brightcove-assigned id, or a customer-generated
+ * reference id. Reference ids are prefixed 'ref:' and the colon
+ * must be preserved unencoded.
+ * @param {string} id
+ * @return {string}
+ */
+function encodeId(id) {
+  /*  */
+  if (id.substring(0, 4) === 'ref:') {
+    return `ref:${encodeURIComponent(id.substring(4))}`;
+  }
+  return encodeURIComponent(id);
+}
+
+/**
+ * @param {string} method
+ * @param {*=} arg
+ * @return {string}
+ */
+function makeMessage(method, arg) {
+  switch (method) {
+    case 'showControls':
+    case 'mute':
+      arg = true;
+      break;
+    case 'hideControls':
+    case 'unmute':
+      arg = false;
+      break;
+    default:
+      break;
+  }
+  return JSON.stringify({
+    'command': methods[method] ?? method,
+    'args': arg,
+  });
+}
+
+/**
+ * Verify required props and throw error if necessary.
+ * @param {string|undefined} account
+ * @return {boolean} true on valid
+ */
+function checkProps(account) {
+  // Perform manual checking as assertion is not available for Bento: Issue #32739
+  if (!account) {
+    displayWarning('account is required for <Brightcove>');
+    return false;
+  }
+  return true;
+}
+
+/**
+ * @param {?string} message
+ */
+function displayWarning(message) {
```

Comment:
FMI: do we use `console` functions instead of `user()`/`dev()` throughout bento?

Reply:
yes

Reply:
nit: simply `warn()` or `console./* OK */log` directly?

Reply:
> nit: simply `warn()` or `console./* OK */log` directly?

I wonder if it's better to just do this going forward. This review comment comes up on every new Bento component that defines `displayWarning` for this purpose, and the reasoning tends to be to not have to `/* OK */` every call. But in reality, the call usually only occurs once. I'll change it here and consider removing `displayWarning` in the existing places as well.